R/FilteredDataset.R

```diff
@@ -407,6 +423,15 @@ FilteredDataset <- R6::R6Class( # nolint
           logger::log_trace("FilteredDataset$server initializing, dataname: { deparse1(dataname) }")
           checkmate::assert_string(dataname)
           shiny::setBookmarkExclude("remove_filters")
+
+          output$filter_count <- renderText(
+            sprintf(
+              "%d filter%s applied",
+              length(self$get_filter_state()),
```

Comment:
I think you need a private method `n_active_filter_states` for both `FilteredDataset` and `MAEFilteredDataset`. In the near future it wont be needed as we plan to unnest these filter-states in the API
```suggestion
              private$n_active_filter_states(),
```

Reply:
So:

```
# how many filters are currently applied
    n_active_filter_states = function() {
      sum(vapply(names(self$get_filter_state()), function(state_name) {
        if (state_name == "subjects") {
          length(self$get_filter_state()[[state_name]])
        } else {
          length(self$get_filter_state()[[state_name]]$subset)
        }
      }, FUN.VALUE = numeric(1)))
    },
```

seems to work?

Reply:
actually I want to move this to FilterStates I think

Reply:
MAE is ugly itself so I'm not judging this. I'm good with the current state (with n_active_filter_states), but if you'd like to move it to filter_states I'm not against this. But I just wonder how are you going to adjust the icon (which is in the filetredDataset) with label produced by FilterStates